modules/ROOT/pages/mtf-metadata-testing.adoc

```diff
@@ -0,0 +1,363 @@
+  public String getCategoryName() {
+    return "ExampleCategory";
+  }
+
+}
+----
+
+.OutputResolver
+[source, java, linenums]
+----
+public class ExampleOutputResolver implements OutputTypeResolver<String> {
+
+  @Override
+  public String getResolverName() {
+    return "exampleResolver";
+  }
+
+  @Override
+  public MetadataType getOutputType(MetadataContext context, String param) throws MetadataResolvingException {
+    if (param == null || param.isEmpty()) {
+      throw new MetadataResolvingException("No metadata available for an empty param", FailureCode.INVALID_METADATA_KEY);
+    }
+    ObjectTypeBuilder objectType = context.getTypeBuilder().objectType();
+    objectType.addField().key("name").value().stringType();
+    objectType.addField().key("age").value().numberType();
+    return objectType.build();
+  }
+
+  @Override
+  public String getCategoryName() {
+    return "ExampleCategory";
+  }
+}
+----
+
+== Test Metadata Keys
+
+The following example validates that the keys have the proper `Display` names:
+
+[source, xml, linenums]
+----
+<mtf:tooling-test name="keysTest">
+    <mtf:get-metadata-keys>
+        <test-connector:with-metadata />
+    </mtf:get-metadata-keys>
+    <mtf:validation>
+        <munit-tools:assert-equals actual="#[payload.firstKey.displayName]" expected="#['First Key']"/>
+        <munit-tools:assert-equals actual="#[payload.secondKey.displayName]" expected="#['Second Key']"/>
+    </mtf:validation>
+</mtf:tooling-test>
+----
+
+== Test a Metadata Type
+
+The following test validates the `Metadata Type` output value returned by the operation:
+
+[source, xml, linenums]
+----
+<mtf:tooling-test name="outputMetadataTest">
+    <mtf:get-output-metadata>
+        <test-connector:with-metadata param="aValue"/>
+    </mtf:get-output-metadata>
+    <mtf:validation>
+        <munit-tools:assert-equals actual="#[payload.format]" expected="#['java']"/>
+        <munit-tools:assert-equals actual="#[payload.fields[0].key.name]" expected="#['name']"/>
+        <munit-tools:assert-equals actual="#[payload.fields[0].model.'type']" expected="#['String']"/>
+        <munit-tools:assert-equals actual="#[payload.fields[1].key.name]" expected="#['age']"/>
+        <munit-tools:assert-equals actual="#[payload.fields[1].model.'type']" expected="#['Number']"/>
+    </mtf:validation>
+</mtf:tooling-test>
+----
+
+== Test an Expected Failure
+
+In the previous example, the module throws a `MetadataResolvingException` error when the parameter is null or empty.
+
+The following test validates that the proper failure code and message are thrown when the parameter is invalid:
+
+[source, xml, linenums]
+----
+<mtf:tooling-test name="failureTest"
+                  expectFailureMessage="No metadata available for an empty param"
+                  expectFailureCode="INVALID_METADATA_KEY">
+    <mtf:get-output-metadata>
+        <test-connector:with-metadata param=""/>
+    </mtf:get-output-metadata>
+</mtf:tooling-test>
+----
+
+
```

Comment:
No See Alsos?

Reply:
No. We typically only add them if there's relevant info to link to. Not a requirement, but might be a nice-to-have?